### ISSUE TITLE
Explicitly support TypeScript 3.6

### ISSUE THREAD

**Issue by SalvatorePreviti** [typescript-eslint/typescript-eslint#935]
At the moment when running eslint I get a warning about typescript version.
The maximum supported is 3.6.0, but there is a minor version 3.6.2 that have some fixes.
This warning should not be issued if there is only a minor revision change, a minor version bump should not issue any warning

**What code were you trying to parse?**

```sh
eslint file.ts
```

**What did you expect to happen?**

No warnings

**What actually happened?**

=============

WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.6.0

YOUR TYPESCRIPT VERSION: 3.6.2

Please only submit bug reports when using the officially supported version.

=============

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.0.0` |
| `TypeScript`                | `3.6.2` |
| `ESLint`                    | `6.3.0` |
| `node`                      | `12` |
| `npm`                       | `6.11.2` |

**Comment by SantoJambit** [comment]
Actually, it says that version <3.6.0 are supported.. so 3.5.x is the maximum. I think this is correct according to the warning. Not sure why 3.6.x is not supported yet, but that's a different issue.

**Comment by JamesHenry** [comment]
Fixed by https://github.com/typescript-eslint/typescript-eslint/pull/916

**Comment by JamesHenry** [comment]
Updated the issue title - as @SantoJambit points out the original description is not accurate

**Comment by bradzacher** [comment]
FYI - typescript does not follow semantic versioning rules.

Whilst they use a 3-number style to version their package, a "minor" bump from 3.5.Y -> 3.6.0 may include (and for the majority of v3 minors - _has included_) breaking changes.

This means that we do not implicitly support "minor" version bumps.

In this instance - there were breaking changes that we cared about (specifically `"constructor"()` is now treated specifically as a constructor instead of a method called constructor).

--

v3.6 was only released on the 28th. This is a community project run by volunteers. Please bear with us as we bump, test and fix to support the minor version.

**Comment by rtatarinov** [comment]
@JamesHenry It's not fixed. I'm using 2.1.0 version and I still see this warning.

**Comment by JamesHenry** [comment]
It is fixed in master, not in 2.1.0.

Please review the release notes and the readme for how our releases work. Users are never blocked, everything they goes into master is released to npm

**Comment by rtatarinov** [comment]
@JamesHenry ok. I don't understand how I can get this fix in my project?

**Comment by JamesHenry** [comment]
From the README:

https://github.com/typescript-eslint/typescript-eslint#package-versions

![image](https://user-images.githubusercontent.com/900523/64201678-3a7c0f00-ce5d-11e9-83fd-199e54f55275.png)

You can install `2.1.1-alpha.4` and it will be in that.

**Comment by garyo** [comment]
I'm running typescript 3.7.3 and still see this multi-line warning in my console on every lint run (but the lint run is fine). Any way to suppress it?

**Comment by bradzacher** [comment]
the supported range is `>=3.2.1 <3.8.0` (has been this for ~2 months).
Are you sure you're using the latest versions of our tooling?

**Comment by garyo** [comment]
Hmm, I guess I wasn't -- I had 1.13. Updated, fixed all the new errors, now things are OK. Thanks.
